packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts

```diff
@@ -202,87 +209,138 @@ export class ExampleGenerator {
-                    maybeParameterExample: examples
-                        .flatMap((example) => example.serviceHeaders)
-                        .find((sh) => sh.name === h.name)
-                })
-            ),
-            endpointHeaders: endpoint.headers.map((h) =>
-                this.generateHttpParameterExample({
-                    parameter: h,
-                    maybeParameterExample: examples
-                        .flatMap((example) => example.endpointHeaders)
-                        .find((sh) => sh.name === h.name)
-                })
-            ),
-            queryParameters: endpoint.queryParameters.map((q) => ({
-                ...this.generateHttpParameterExample({
-                    parameter: q,
-                    maybeParameterExample: examples
-                        .flatMap((example) => example.queryParameters)
-                        .find((qp) => qp.name === q.name)
-                }),
-                shape: q.allowMultiple ? ExampleQueryParameterShape.exploded() : ExampleQueryParameterShape.single()
-            })),
-            request:
-                endpoint.requestBody !== undefined
-                    ? this.generateRequestBodyExample({
-                          requestBody: endpoint.requestBody,
-                          maybeExampleRequest: examples
-                              .map((example) => example.request)
-                              .filter((example) => example != null)[0]
-                      })
-                    : undefined,
+    ): Omit<ExampleEndpointCall, "id" | "response"> | undefined {
+        const result: Omit<ExampleEndpointCall, "id" | "response" | "url"> = {
             name: undefined,
-            docs: undefined
+            docs: undefined,
+            rootPathParameters: [],
+            servicePathParameters: [],
+            endpointPathParameters: [],
+            serviceHeaders: [],
+            endpointHeaders: [],
+            queryParameters: [],
+            request: undefined
         };
-        const allPathParameters = [
-            ...exampleWithoutUrl.rootPathParameters,
-            ...exampleWithoutUrl.servicePathParameters,
-            ...exampleWithoutUrl.endpointPathParameters
+
+        const examples = [
+            ...endpoint.userSpecifiedExamples.map((userSpecified) => userSpecified.example).filter(isNonNullish)
         ];
-        const pathParameterStringsByName = this.examplePathParametersToRecord(allPathParameters);
+
+        /***** Path Parameter Examples *****/
+
+        for (const pathParameter of rootPathParameters) {
+            const example = this.generatePathParameterExample({
+                pathParameter,
+                maybePathParameterExample: examples
+                    .flatMap((example) => example.rootPathParameters)
+                    .find((sp) => sp.name === pathParameter.name)
+            });
+            if (example == null) {
+                return undefined;
```

Comment:
Why return here? vs just adding if not null? Presumably you'd want to finish the loop and continue on?

Reply:
that file is going to completely deleted now